pykafka/protocol.py

```diff
@@ -212,13 +217,13 @@ def __len__(self):
 
     @classmethod
     def decode(self, buff, msg_offset=-1, partition_id=-1):
-        (crc, protocol_version) = struct_helpers.unpack_from('iB', buff, 0)
-        offset = 5
+        (crc, protocol_version, attr) = struct_helpers.unpack_from('iBB', buff, 0)
```

Comment:
Good catch, I totally missed this yesterday and it's probably what was driving me insane.